org.adempiere.base/src/org/compiere/wf/MWFNextCondition.java

```diff
@@ -342,5 +364,40 @@ public MWFNextCondition markImmutable() {
 		makeImmutable();
 		return this;
 	}
+	
+	/**
+	 * Executes SQLStatement of Record and returns result as Object
+	 * @param Record_ID
+	 * @return SQL statement execution result
+	 * @throws Exception
+	 */
+	private Object getSqlResult(int Record_ID) throws Exception {
+		String sqlStatement = (String) get_Value("SQLStatement");
+		String tableName = getAD_Column().getAD_Table().getTableName();
+		String pkName = tableName + "_ID"; // primary key name
+
+		Object result = null;
+
+		String resultSql = String.format("SELECT (%s) FROM %s WHERE %s = %d", sqlStatement, tableName, pkName, Record_ID);
+
+		PreparedStatement pstmt = null;
+		ResultSet rs = null;
+
+		try {
+			pstmt = DB.prepareStatement(resultSql, null);

```

Comment:
for better security, call the DB.prepareStatement method that make sure the resultset is readonly.